source/particle/output/ascii.cc

```diff
@@ -0,0 +1,140 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/output/ascii.h>
+
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Output
+    {
+      template <int dim>
+      ASCIIOutput<dim>::ASCIIOutput()
+        :
+        file_index(0)
+      {}
+
+      template <int dim>
+      std::string
+      ASCIIOutput<dim>::output_particle_data(const std::multimap<types::LevelInd, Particle<dim> > &particles,
+                                             const std::vector<std::pair<std::string, unsigned int> > &property_component_list,
+                                             const double /*time*/)
+      {
+        const std::string output_file_prefix = "particle-" + Utilities::int_to_string (file_index, 5);
+        const std::string output_path_prefix = this->get_output_directory() + output_file_prefix;
+        const std::string full_filename = output_path_prefix + "." + Utilities::int_to_string(Utilities::MPI::this_mpi_process(this->get_mpi_communicator()), 4) + ".txt";
+        std::ofstream output (full_filename.c_str());
+        if (!output)
+          std::cout << "ERROR: proc " << Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) << " could not create " << full_filename << std::endl;
```

Comment:
make this an `AssertThrow` of some kind so that we get a location.

Or is your intention to continue even if you can't write (which is what you do here; though you could as well put a `return` into the `if` clause).